uint8 msg_type
uint64 event_type
uint8 result